🐛 BUG: Sitemap 0.20 introduces `ERR_MODULE_NOT_FOUND`

### What version of `astro` are you using?

1.0.0-beta.48

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

NPM

### What operating system are you using?

Windows 11

### Describe the Bug

When running `npm run build` with the sitemap package it'll throw the following error:

`Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'C:\Users\erik\Downloads\sites\astro-blog-portfolio\node_modules\@astrojs\sitemap\dist\generate-sitemap' imported from C:\Users\erik\Downloads\sites\astro-blog-portfolio\node_modules\@astrojs\sitemap\dist\index.js`

This happens on a manual install as well as the Astro Add CLI command.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-de1e1j?file=package.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/gold-walls-pretend.md
+++ b/.changeset/gold-walls-pretend.md
@@ -1,0 +1,5 @@
+---
+'@astrojs/sitemap': patch
+---
+
+Fixes the last build

--- a/packages/integrations/sitemap/package.json
+++ b/packages/integrations/sitemap/package.json
@@ -28,7 +28,8 @@
   "scripts": {
     "build": "astro-scripts build \"src/**/*.ts\" && tsc",
     "build:ci": "astro-scripts build \"src/**/*.ts\"",
-    "dev": "astro-scripts dev \"src/**/*.ts\""
+    "dev": "astro-scripts dev \"src/**/*.ts\"",
+    "test": "mocha --timeout 20000"
   },
   "dependencies": {
     "sitemap": "^7.1.1",

--- a/packages/integrations/sitemap/src/config-defaults.ts
+++ b/packages/integrations/sitemap/src/config-defaults.ts
@@ -1,4 +1,4 @@
-import type { SitemapOptions } from './index';
+import type { SitemapOptions } from './index.js';
 
 export const SITEMAP_CONFIG_DEFAULTS: SitemapOptions & any = {
 	entryLimit: 45000,

--- a/packages/integrations/sitemap/src/generate-sitemap.ts
+++ b/packages/integrations/sitemap/src/generate-sitemap.ts
@@ -1,5 +1,5 @@
-import type { SitemapItem, SitemapOptions } from './index';
-import { parseUrl } from './utils/parse-url';
+import type { SitemapItem, SitemapOptions } from './index.js';
+import { parseUrl } from './utils/parse-url.js';
 
 const STATUS_CODE_PAGE_REGEXP = /\/[0-9]{3}\/?$/;
 

--- a/packages/integrations/sitemap/src/index.ts
+++ b/packages/integrations/sitemap/src/index.ts
@@ -8,9 +8,9 @@ import {
 import { fileURLToPath } from 'url';
 import { ZodError } from 'zod';
 
-import { generateSitemap } from './generate-sitemap';
-import { Logger } from './utils/logger';
-import { validateOptions } from './validate-options';
+import { generateSitemap } from './generate-sitemap.js';
+import { Logger } from './utils/logger.js';
+import { validateOptions } from './validate-options.js';
 
 export type ChangeFreq = EnumChangefreq;
 export type SitemapItem = Pick<

--- a/packages/integrations/sitemap/src/schema.ts
+++ b/packages/integrations/sitemap/src/schema.ts
@@ -1,6 +1,6 @@
 import { EnumChangefreq as ChangeFreq } from 'sitemap';
 import { z } from 'zod';
-import { SITEMAP_CONFIG_DEFAULTS } from './config-defaults';
+import { SITEMAP_CONFIG_DEFAULTS } from './config-defaults.js';
 
 const localeKeySchema = z.string().min(1);
 

--- a/packages/integrations/sitemap/src/validate-options.ts
+++ b/packages/integrations/sitemap/src/validate-options.ts
@@ -1,6 +1,6 @@
 import { z } from 'zod';
-import type { SitemapOptions } from './index';
-import { SitemapOptionsSchema } from './schema';
+import type { SitemapOptions } from './index.js';
+import { SitemapOptionsSchema } from './schema.js';
 
 // @internal
 export const validateOptions = (site: string | undefined, opts: SitemapOptions) => {

--- a/packages/integrations/sitemap/test/smoke.test.js
+++ b/packages/integrations/sitemap/test/smoke.test.js
@@ -1,0 +1,3 @@
+import '../dist/index.js';
+
+// Just a smoke test, this would fail if there's a problem.

